.github/workflows/black_n_pylint.yml

```diff
@@ -0,0 +1,63 @@
+# This workflow runs pylint and updates badge
+
+name: pylint
+
+on:
+  pull_request:
+    branches:
+      - "main"
+      - "develop"
+    paths-ignore:
+      - "docs/**"
+
+jobs:
+  black-n-pylint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.x"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install black lyncs_setuptools[pylint]
+          pip install lyncs_io
```

Comment:
Installs only what's required and avoids installing the extras as that would require installing MPI too.

Reply:
I see your problem, but then pylint will not find content of modules that are not installed by default. For MPI you needed to install it via sudo, before installing the dependencies.
`sudo apt-get install -y libopenmpi-dev openmpi-bin`